applicationset/generators/git.go

```diff
@@ -58,12 +61,36 @@ func (g *GitGenerator) GenerateParams(appSetGenerator *argoprojiov1alpha1.Applic
 
 	noRevisionCache := appSet.RefreshRequired()
 
+	var project string
+	if strings.Contains(appSet.Spec.Template.Spec.Project, "{{") {
+		project = appSetGenerator.Git.Template.Spec.Project
+	} else {
+		project = appSet.Spec.Template.Spec.Project
+	}
+
+	log.Info("===============project=================")
+	log.Info(project)
+
+	appProject := &argoprojiov1alpha1.AppProject{}
+	if err := client.Get(context.TODO(), types.NamespacedName{Name: appSet.Spec.Template.Spec.Project, Namespace: appSet.Namespace}, appProject); err != nil {
+		return nil, fmt.Errorf("error getting project %s: %w", project, err)
+	}
+
+	log.Infof("===============GPG Enabled : %v=================", gpg.IsGPGEnabled())
+	// we need to verify the signature on the Git revision if GPG is enabled
+	verifyCommit := false
```

Comment:
can we just change it to verifyCommit := appProject.Spec.SignatureKeys != nil && len(appProject.Spec.SignatureKeys) > 0 && gpg.IsGPGEnabled()

Reply:
Sure.. Tried keeping it simple and matching to what we have for applications today.